compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,301 @@
+
+        self.resolution = resolution
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+
+        section = config['mesh']
+
+        # read the resolution from the .cfg file at runtime
+        resolution = section.getint('resolution')
+
+        # check if the resolution has been changed since the `compass setup`
+        # command was run
+        if self.resolution != resolution:
+            warnings.warn(f'Resolution was set at {self.resolution:2d}km when'
+                          f' `compass setup` was called. Since then the'
+                          f' resolution in the configuration file has been'
+                          f' changed to {resolution:2d}km. Changing resolution'
+                          f' at runtime is not supported. Change the'
+                          f' resolution value in the configuration file within'
+                          f' the python module and rerun the `compass setup`'
+                          f' command in order to create a mesh at a resolution'
+                          f' of {resolution:2d}km')
+
+        ds_mesh = make_planar_hex_mesh(nx=self.nx, ny=self.ny, dc=self.dc,
+                                       nonperiodic_x=True,
+                                       nonperiodic_y=True)
+
+        ds_mesh = cull(ds_mesh, logger=logger)
+        ds_mesh = convert(ds_mesh, logger=logger)
+
+        # custom function to shift the orgin for the MISMIP+ experiments
+        ds_mesh = center_trough(ds_mesh)
+
+        write_netcdf(ds_mesh, 'mpas_grid.nc')
+
+        # using `.get(...)`, instead of `.getint(...)` since variable needs
+        # to be string anyway for subprocess call
+        levels = section.get('levels')
+
+        args = ['create_landice_grid_from_generic_MPAS_grid.py',
+                '-i', 'mpas_grid.nc',
+                '-o', 'landice_grid.nc',
+                '-l', levels,
+                '--diri',
+                '--thermal']
+
+        check_call(args, logger)
+
+        make_graph_file(mesh_filename='landice_grid.nc',
+                        graph_filename='graph.info')
+
+        # create the initial conditions for the MISMIP+ spinup expr
+        _setup_MISMPPlus_IC(config, logger,
+                            filename='landice_grid.nc')
+
+
+def center_trough(ds_mesh):
+    """
+    Shift the orgin so that the bed trough is center about the Y-axis and the
+    X-axis is shifted all the way to the left.
+
+    Parameters
+    ----------
+    ds_mesh : xr.core.dataset.Datatset
+        The mesh to be shifted
+    """
+
+    # Get the center of y-axis (i.e. half distance)
+    yCenter = (ds_mesh.yCell.max() + ds_mesh.yCell.min()) / 2.
+
+    # Shift x-axis so that the x-origin is all the way to the left
+    xShift = -1.0 * ds_mesh.xCell.min()
+    # Shift y-axis so that it's centered about the MISMIP+ bed trough
+    yShift = 4.e4 - yCenter
```

Comment:
@andrewdnolan , these shifts are slightly different from the old MISMIP+ test case, right?  If I followed things, these shifts are more sensible.  However, @trhille, be aware that if you interpolate the state from our archived MISMIP+ 2km state file, the coordinate system of these new meshes will be slightly different unless you do extra work to align them.